### ISSUE TITLE
bundle install or update of github-pages fails

### ISSUE THREAD

**Issue by Merganser2** [github/pages-gem#819]
### This issue affects

- [ ] Building sites locally

### What did you do (e.g., steps to reproduce)
- Installed Ruby31-x64 on Windows 11
- gem update bundler (nothing to update)
- bundle init
- Added "gem 'github-pages', group: :jekyll_plugins" to Gemfile
- bundle install

### What did you expect to happen?
- githubpages-gem to be installed

### What happened instead?
An error occurred while installing unf_ext (0.0.8), and Bundler cannot continue.

In Gemfile:
  github-pages was resolved to 223, which depends on
    github-pages-health-check was resolved to 1.17.9, which depends on
      dnsruby was resolved to 1.61.9, which depends on
        simpleidn was resolved to 0.2.1, which depends on
          unf was resolved to 0.1.4, which depends on
            unf_ext

### Additional information

Further up there were some other errors making me think I missed some setup step:

Ruby31-x64/lib/ruby/3.1.0 -r
./siteconf20220221-7320-1auhwk.rb extconf.rb
checking for -lstdc++... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

**Comment by parkr** [comment]
More information should be present in the `mkmf.log` and would be helpful to have.

It looks like pre-compiled binaries for unf_ext exist for Ruby 3.0: https://github.com/knu/ruby-unf_ext/blob/master/CHANGELOG.md

Perhaps you need to install Ruby 3.0 instead of Ruby 3.1 and try again until unf_ext releases a new version with support for Ruby 3.1.

**Comment by parkr** [comment]
Yes, it looks like an issue with Ruby 3.1 on Windows: https://github.com/knu/ruby-unf_ext/issues/65. It's not yet supported. Subscribe to that issue for updates.

**Comment by Merganser2** [comment]
Yes I got it to work with Ruby 3.0.3 on Windows 11, had to also install webrick gem (chose 1.7 for some reason). Apologies for missing your suggestion about mkmf.log, I uninstalled before copying it. I do remember that it was a very short entry that looked something like something from a cpp header (maybe, if I recall correctly). Just a few lines. Thanks for promptly helping me with this workaround. I am wondering if this should be documented somewhere.
